src/tests/multihost/ad/test_access_control.py

```diff
@@ -0,0 +1,503 @@
+            1. Set simple_allow_user to $ symbol in sssd.conf and restart sssd
+            2. Log in as AD user
+        :expectedresults:
+            1. Changes should reflected in restarted sssd
+            2. Any AD-User log in should be denied
+        """
+        tools = sssdTools(multihost.client[0])
+        (aduser, _) = create_aduser_group
+        domain_name = tools.get_domain_section_name()
+        dom_section = f'domain/{domain_name}'
+        sssd_params = {'access_provider': 'simple',
+                       'simple_allow_users': '$'}
+        tools.sssd_conf(dom_section, sssd_params, action='add')
+        tools.clear_sssd_cache()
+        client_hostname = multihost.client[0].sys_hostname
+        client = pexpect_ssh(client_hostname, f'{aduser}@{domain_name}', 'Secret123', debug=False)
+        try:
+            client.login()
+        except SSHLoginException:
+            pytest.fail(f'{aduser} failed to login')
+        except pexpect.EOF as err:
+            log_str = multihost.client[0].get_file_contents('/var/log/secure').decode('utf-8')
+            patt = re.compile(r'Access.*denied for user')
+        else:
+            client.logout()
+        assert patt.findall(log_str), 'An Access denied log is absent in /var/log/secure'
+
+    @staticmethod
+    def test_simple_allow_user_to_invalid_user(multihost, create_aduser_group, backupsssdconf):
+        """
+        :title: Set simple_allow_user to an invalid user
+        :id: fdde80ab-1364-4953-893e-62f7c3377ff8
+        :description: Setting simple_allow_user to an invalid user, with this sssd would deny log in of all users
+        :steps:
+            1. Set simple_allow_user to non existing user in sssd.conf and restart sssd
+            2. Log in as AD user
+        :expectedresults:
+            1. Changes should reflected in restarted sssd
+            2. User log in should be denied
+        """
+        tools = sssdTools(multihost.client[0])
+        (aduser, _) = create_aduser_group
+        domain_name = tools.get_domain_section_name()
+        dom_section = f'domain/{domain_name}'
+        sssd_params = {'access_provider': 'simple',
+                       'simple_allow_users': f'nonuser@{domain_name}'}
+        tools.sssd_conf(dom_section, sssd_params, action='add')
+        tools.clear_sssd_cache()
+        client_hostname = multihost.client[0].sys_hostname
+        client = pexpect_ssh(client_hostname, f'{aduser}@{domain_name}', 'Secret123', debug=False)
+        try:
+            client.login()
+        except SSHLoginException:
+            pytest.fail(f'{aduser} failed to login')
+        except pexpect.EOF:
+            log_str = multihost.client[0].get_file_contents('/var/log/secure').decode('utf-8')
+            patt = re.compile(r'Access.*denied for user')
+            assert patt.findall(log_str), 'An Access denied log is absent in /var/log/secure'
+        else:
+            (stdout, _) = client.command(f'id {aduser}@{domain_name}')
```

Comment:
This line does not seem to be needed for anything as it is not part of the steps, nor it is evaluated.

Reply:
taken care of with using function for log in